aten/src/ATen/cuda/Sleep.cu

```diff
@@ -7,12 +7,21 @@ namespace at {
 namespace cuda {
 namespace {
 __global__ void spin_kernel(int64_t cycles) {
+  // Few AMD specific GPUs have different clock intrinsic
+#if defined(__GFX11__) && defined(USE_ROCM) && !defined(__CUDA_ARCH__)
```

Comment:
Do we need both `USE_ROCM` and `__CUDA_ARCH__`?

Reply:
To make the sure the code is enabled only for ROCm.